docs/guides/jupyter.rst

```diff
@@ -0,0 +1,363 @@
+
+To further customize the rendering process among other things,
+refer to the nbsphinx_ or MyST-NB_ documentation.
+
+Rendering interactive widgets
+-----------------------------
+
+Widgets are eventful python objects that have a representation in the browser
+and that you can use to build interactive GUIs for your notebooks.
+Basic widgets using `ipywidgets`_ include controls like sliders, textboxes, and buttons,
+and more complex widgets include interactive maps, like the ones provided by `ipyleaflet`_.
+
+.. _ipywidgets: https://ipywidgets.readthedocs.io/
+.. _ipyleaflet: https://ipyleaflet.readthedocs.io/
+
+You can embed these interactive widgets on HTML Sphinx documentation.
+For this to work, it's necessary to *save the widget state*
+before generating the HTML documentation,
+otherwise the widget will appear as empty.
+Each editor has a different way of doing it:
+
+- The classical Jupyter Notebook interface
+  provides a "Save Notebook Widget State" action in the "Widgets" menu,
+  :ref:`as explained in the ipywidgets
+  documentation <ipywidgets:/embedding.md#embedding-widgets-in-html-web-pages>`.
+  You need to click it before exporting your notebook to HTML.
+- JupyterLab provides a "Save Widget State Automatically" option in the "Settings" menu.
+  You need to leave it checked so that widget state is automatically saved.
+- In Visual Studio Code `it's not possible to save the widget
+  state <https://github.com/microsoft/vscode-jupyter/issues/4404>`_
+  at the time of writing.
+
+.. figure:: /_static/images/guides/jupyterlab-save-widget-state.png
+   :width: 30%
+   :align: center
+   :alt: JupyterLab option to save the interactive widget state automatically
+
+   JupyterLab option to save the interactive widget state automatically
+
+For example, if you create a notebook with a simple
+:ref:`ipywidgets:/examples/widget list.ipynb#intslider`
+widget from ipywidgets and save the widget state,
+the slider will render correctly in Sphinx.
+
+.. figure:: /_static/images/guides/widget-html.gif
+   :width: 80%
+   :align: center
+   :alt: Interactive widget rendered in HTML by Sphinx
+
+   Interactive widget rendered in HTML by Sphinx
+
+To see more elaborate examples:
+
+- `ipyleaflet`_ provides several widgets for interactive maps,
+  and renders live versions of them `in their
+  documentation <https://ipyleaflet.readthedocs.io/en/latest/api_reference/velocity.html>`_.
+- `PyVista <https://docs.pyvista.org/>`_ is used for scientific 3D visualization
+  with several interactive backends and `examples in their
+  documentation <https://docs.pyvista.org/index.html#maps-and-geoscience>`_ as well.
+
+.. warning::
+
+   Although widgets themselves can be embedded in HTML as shown above,
+   :doc:`events <ipywidgets:examples/Widget Events>`
+   require a backend (kernel) to execute.
+   Therefore, ``@interact``, ``.observe``, and related functionalities relying on them
+   will not work as expected.
+
+.. _notebook-galleries:
```

Comment:
```suggestion
.. warning::

   Some interactive cell outputs expect certain **javascript dependencies** to be loaded in order to correctly display the output (e.g., [`plotly.js`](https://github.com/plotly/plotly.js/) for Plotly outputs).
   In Jupyter interfaces, these libraries are usually loaded via extensions, but building your documentation with notebooks requires you to lead the libraries yourself.

.. _notebook-galleries:
```

Reply:
I'd also cross-link to some documentation about how people can load their own JS libraries.

Reply:
I'm not sure if it's worth adding this warning. I tried creating a project from scratch with `ipywidgets` and it worked without adding anything, and on the other hand Plotly has some rendering modes that either bundle or load using `require.js` the necessary code https://plotly.com/python/renderers/. But of course there might be other cases where some action is actually needed?

Reply:
Maybe

```
.. note::

   If your widgets need some additional JavaScript libraries,
   you can add them using ``Sphinx.add_js_file``
```

?

Reply:
hmm - that sounds like a reasonable first step, though you might get users who are confused because their JS works in JupyterLab but not in the Sphinx HTML (Jupyter Book has a ton of issues opened around this)

Reply:
Done!